### PR TITLE
[RSDK-6913] & [RSDK-6914] ServoError+BaseError+GenericError

### PR DESCRIPTION
**Pull request by npmenard**

### CODE REVIEW

File: micro-rdk/src/esp32/mod.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ pub mod exec;
 pub mod hcsr04;
 pub mod i2c;
 pub mod pin;
+#[cfg(feature = "builtin-components")]
```
Comment (gvaradarajan) [inline micro-rdk/src/esp32/mod.rs:18]: are we sure about this one? It's really more of a peripheral than it is a component

**Review by gvaradarajan — APPROVED** [review]
LGTM

File: micro-rdk/src/esp32/mod.rs (hunk 1 above)
Comment (npmenard) [reply micro-rdk/src/esp32/mod.rs:18]: it gives clippy error with "--features esp32 --no-default-features"

Comment (gvaradarajan) [reply micro-rdk/src/esp32/mod.rs:18]: oh I see, nvm

**Review by acmorrow — APPROVED** [review]
LGTM

**Pull request merged**